### DISCUSSION TITLE
Make all windows top most (global `window created` hook)

### DISCUSSION THREAD

**Discussion by blahetal** [AvaloniaUI/Avalonia#11714] [General]
Hi,
how can make all windows (I am using MVVM) topmost without setting `window.topmost` property on every window during application lifetime (it is Windows desktop application)?
Does Avalonia has a global hook or an event that would be called when a window is created so that would be the place (and the only place in the whole application) where I could set the topmost flag? Otherwise I will need to tweak every window and set the flag accordingly.

Thanks,
Libor

**Comment by timunie** [comment]
isn't that a bindable property? https://reference.avaloniaui.net/api/Avalonia.Controls/WindowBase/10A1BACF

**Reply by blahetal** [reply]
It is but I would need to bind it on every window which I do not want to do. So something like a window created hook (if there is one) would allow me to set it from one place.

**Reply by timunie** [reply]
As it is a styled property, you can set it in Style I thought. 

```xml
<Style Selector=":is(WindowBase)">
   <Setter Property="TopMost" Value="True" />
</Style>
```

otherwise, you may try to see if `ClassicDektopLifetime` has such an event you need.

**Reply by blahetal** [reply]
thanks, I will try it. Sometimes I forget how are the styles powerful and try to solve everything in code

**Comment by blahetal** [comment]
Closing
